examples/apps/src/ble_nus_peripheral.rs

```diff
@@ -0,0 +1,133 @@
+/* Nordic Uart Service (NUS) peripheral example */
+use embassy_futures::join::join3;
+use embassy_sync::blocking_mutex::raw::NoopRawMutex;
+use embassy_time::{Duration, Timer};
+use heapless::Vec;
+use trouble_host::prelude::*;
+
+/// Size of L2CAP packets (ATT MTU is this - 4)
+const L2CAP_MTU: usize = 251;
+
+/// Max number of connections
+const CONNECTIONS_MAX: usize = 1;
+
+/// Max number of L2CAP channels.
+const L2CAP_CHANNELS_MAX: usize = 2; // Signal + att
+
+const MAX_ATTRIBUTES: usize = 32;
+
+pub const MTU: usize = 120;
+// Aligned to 4 bytes + 3 bytes for header
+pub const ATT_MTU: usize = MTU + 3;
+
+type Resources<C> = HostResources<C, CONNECTIONS_MAX, L2CAP_CHANNELS_MAX, L2CAP_MTU>;
+
+// GATT Server
+#[gatt_server]
+struct Server {
+    nrf_uart: NrfUartService,
+}
+
+// NRF UART Service
+#[gatt_service(uuid = "6E400001-B5A3-F393-E0A9-E50E24DCCA9E")]
+struct NrfUartService {
+    #[characteristic(uuid = "6E400002-B5A3-F393-E0A9-E50E24DCCA9E", write)]
+    rx: Vec<u8, ATT_MTU>,
+
+    #[characteristic(uuid = "6E400003-B5A3-F393-E0A9-E50E24DCCA9E", notify)]
+    tx: Vec<u8, ATT_MTU>,
+}
+
+pub async fn run<C>(controller: C)
+where
+    C: Controller,
+{
+    let address = Address::random([0x41, 0x5A, 0xE3, 0x1E, 0x83, 0xE7]);
+    info!("Our address = {:?}", address);
+
+    let mut resources = Resources::new(PacketQos::None);
+    let (stack, peripheral, _, runner) = trouble_host::new(controller, &mut resources)
+        .set_random_address(address)
+        .build();
+
+    let mut table: AttributeTable<'_, NoopRawMutex, MAX_ATTRIBUTES> = AttributeTable::new();
+
+    // Generic Access Service (mandatory)
+    let id = b"Trouble Example Device";
+    let mut svc = table.add_service(Service::new(0x1800));
+    let _ = svc.add_characteristic_ro(0x2a00, id);
+    svc.build();
+
+    // Generic attribute service (mandatory)
+    table.add_service(Service::new(0x1801));
```

Comment:
Just tagging you here @petekubiak and @jamessizeland, I wonder if it would make sense to have the macros generate these mandatory services? Iirc the old softdevice does that internally.

Reply:
We discussed this when working on the last PR, and are definitely keen to implement something along these lines at some point.
I'm not an expert in BLE so excuse if this is a silly question, but my understanding is that these services are associated with GAP for a generic BLE device. Is this something we would always want included, or instead an optional feature if the user wants to implement GAP in their application?

Reply:
According to the BLE spec (https://www.bluetooth.com/wp-content/uploads/Files/Specification/HTML/Core-54/out/en/host/generic-access-profile.html#UUID-8942f34d-80a7-1f7f-8f92-c8ee39aed116_informaltable-idm13358906307630), they are mandatory ("The GATT Server shall contain the GAP service as defined in the GAP Service Requirements in Table 12.1. A device shall have only one instance of the GAP service in the GATT Server. The GAP service is a GATT based service with the service UUID as «GAP Service» defined in [Assigned Numbers](https://www.bluetooth.com/specifications/assigned-numbers).")

So I don't think it should be optional.

Reply:
Aha, thanks for the info! Turns out I hadn't properly remembered what @jamessizeland had told me - he had pointed out that devices could present as various different types of device depending on the way in which the GAP service was set up. Long story short, this is definitely functionality we'd like to add to the macro yes :)

Reply:
Agree we should add it either to the macro or as a builder. I'll sketch out some options tomorrow to see what's most ergonomic.

I'll open an issue to discuss.

Reply:
https://github.com/embassy-rs/trouble/issues/142

Reply:
This is completed and merged now, should no longer be blocking this for you @plaes :)